src/client/main.tsx

```diff
@@ -24,22 +26,45 @@ export class Main extends React.Component<{}, ClientState> {
       stage: Stages.SPLASH,
       numPlayersInLobby: 0,
     };
+    let pathName = window.location.pathname.substring(1); // Cutting off preceding '/'
+    if (pathName.length === GAME_CODE_LENGTH) {
+      // We are reserving all 5 character urls as game urls
+      this.socketInit(pathName);
+      this.state.stage = Stages.LOADING;
+    }
+  }
+
+  createPrivateGame() {
+    this.createLobby('createPrivate');
+  }
+
+  joinRandomGame() {
+    this.createLobby('join');
+  }
+
+  createLobby(endpoint:string) {
```

Comment:
I think it'd be better to have an `onSubmit` callback in the `Splash` component to allow changing the state to loading, and then put all of this `createLobby` logic in the `Splash` component itself.

Probably better to switch `LOADING` to `LOBBY` too at this point.